### PR TITLE
fix: refactor to introduce photo pipeline + UI improvement + bug fixes (hopefully)

### PR DESCRIPTION
**Pull request by BoyeGuillaume**

### CODE REVIEW

File: app/src/main/java/com/github/geohunt/app/ui/components/CircleLoadingAnimation.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.scale
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.layout.layout
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun CircleLoadingAnimation(
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/CircleLoadingAnimation.kt:24]: Neat animation!

File: app/src/main/java/com/github/geohunt/app/ui/components/challengecreation/CreateChallengeViewModel.kt [hunk 2]
```diff
@@ -28,106 +30,42 @@ class CreateChallengeViewModel(
...
-
-    private val _location : MutableStateFlow<Location?> = MutableStateFlow(null)
-    val location : StateFlow<Location?> = _location
-
-    private val _expirationDate: MutableStateFlow<LocalDate?> = MutableStateFlow(null)
-    val expirationDate: StateFlow<LocalDate?> = _expirationDate
-
     private val _selectedDifficulty: MutableStateFlow<Challenge.Difficulty> =
-        MutableStateFlow(Challenge.Difficulty.MEDIUM)
+    MutableStateFlow(Challenge.Difficulty.MEDIUM)
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/challengecreation/CreateChallengeViewModel.kt:34]: maybe put it in the same line or indent it?

File: app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt [hunk 3]
```diff
@@ -217,7 +217,7 @@ fun NavigationController(
             ClaimChallenge(
                 cid = cid,
                 onFailure = onFailure,
-                onClaimSubmitted = {
+                onSuccess = {
                     navController.popBackStack()
                     navController.navigate("challenge-view/$cid")
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:222]: Maybe reference to HiddenRoute.ChallengeView if it exists? or create it! Maybe making the $cid as an argument to the enum could be possible? Idk. Just make it `${HiddenRoute.ChallengeView.route}/$cid` ?

File: app/src/main/java/com/github/geohunt/app/ui/components/utils/intents/IntentsMocking.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
+package com.github.geohunt.app.ui.components.utils.intents
+
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.compose.runtime.Composable
+
+
+// Once upon a time, orpheus descended to hell with only a lyre. Reading
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/utils/intents/IntentsMocking.kt:7]: bien dit!

File: app/src/androidTest/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallengeTest.kt [hunk 5]
```diff
@@ -49,56 +68,90 @@ class BountyClaimChallengeTest {
...
+                            MockBountyClaimRepository()
+                        )
+                    }
+                    val state = vm.challenge.collectAsState()
+                    if (state.value != null) {
+                        awaitThingy.complete(Unit)
+                    }
+
+                    BountyClaimChallenge(
+                        bid = "<NaN>",
```
Comment (Maeeen) [inline app/src/androidTest/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallengeTest.kt:130]: Not the best name, but alright

**Review by Maeeen — APPROVED** [review]
Overall very good code! This makes the whole codebase easier to read.

Really nice work, I am surprised 
image

File: app/src/main/java/com/github/geohunt/app/ui/components/CircleLoadingAnimation.kt (hunk 1 above)
Comment (BoyeGuillaume) [reply app/src/main/java/com/github/geohunt/app/ui/components/CircleLoadingAnimation.kt:24]: Thanks man 😄

File: app/src/main/java/com/github/geohunt/app/ui/components/challengecreation/CreateChallengeViewModel.kt (hunk 2 above)
Comment (BoyeGuillaume) [reply app/src/main/java/com/github/geohunt/app/ui/components/challengecreation/CreateChallengeViewModel.kt:34]: Oupsis

File: app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt (hunk 3 above)
Comment (BoyeGuillaume) [reply app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:222]: Sure !!

**Pull request merged**